libdnf/rpm/package_query.cpp

```diff
@@ -2148,6 +2148,30 @@ static void add_n_first_to_map(
     }
 }
 
+static void add_duplicates_to_map(
+    libdnf::solv::Pool & pool,
+    libdnf::solv::SolvMap & result,
+    libdnf::solv::IdQueue & samename,
+    int start_block,
+    int stop_block) {
+    Solvable * s_first = nullptr;
+    Solvable * s_second = nullptr;
+    for (int pos_first = start_block; pos_first < stop_block; ++pos_first) {
+        Id id_first = samename[pos_first];
+        s_first = pool.id2solvable(id_first);
+        for (int pos_second = start_block; pos_second < stop_block; ++pos_second) {
+            Id id_second = samename[pos_second];
+            s_second = pool.id2solvable(id_second);
+            if ((s_first->evr == s_second->evr) && (s_first->arch != s_second->arch)) {
```

Comment:
All packages in the block should have the same arch. Why is `&& (s_first->arch != s_second->arch)` needed?

Reply:
It is not, I forgot that the arch check was moved. :+1:  :+1: 

Since thanks to your other suggestion we no longer compare each solvable to it self I think we can drop the condition entirely. If there somehow was a package installed twice with the same `name`, `arch` and even `evr` I think we would want it in the query. Packages with the same `nevra` but different `id` surely are duplicates in general for the `filter_duplices()`.

What do you think?